File: HelpDesk-BE/HelpdeskSystem.Domain/Entities/Notification.cs
using HelpdeskSystem.Common.Enums;
using HelpdeskSystem.Domain.Common;

namespace HelpdeskSystem.Domain.Entities;

public class Notification : BaseEntity
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public int? TicketId { get; set; }
    public NotificationType Type { get; set; }
    public string Message { get; set; } = string.Empty;
    public bool IsRead { get; set; } = false;

    // Navigation properties
    public User User { get; set; } = null!;
    public Ticket? Ticket { get; set; }
}

File: HelpDesk-FE-React/src/interfaces/notification.ts
import type { NotificationType } from "../types/notification";

export interface NotificationResponse {
    id: number;
    ticketId: number | null;
    type: NotificationType;
    message: string;
    isRead: boolean;
    createdAt: string;
}

export interface NotificationRequestFilter {
    page?: number;
    pageSize?: number;
}